spec/latest/core/core.jsonld

```diff
@@ -505,6 +511,55 @@
       "domain": "hydra:IriTemplateMapping",
       "range": "xsd:string",
       "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:returnsHeader",
+      "@type": "rdf:Property",
+      "label": "returns header",
+      "comment": "Name of the header returned by the operation.",
+      "domain": "hydra:Operation",
+      "range": "xsd:string",
+      "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:expectsHeader",
+      "@type": "rdf:Property",
+      "label": "expects header",
+      "comment": "Specification of the header expected by the operation.",
+      "domain": "hydra:Operation",
+      "rangeIncludes": ["hydra:Header", "hydra:HeaderTemplate"],
+      "vs:term_status": "testing"
+    },
+    {
+      "@id": "hydra:Template",
+      "@type": "rdfs:Class",
+      "label": "Template",
+      "comment": "Some abstract template.",
+      "vs:term_status": "testing"
+    },

```

Comment:
I'd like to challenge the very need for an `abstract` template. It sounds like smuggling OOP terminology into the vocabulary which strikes me as awkward

Reply:
I'm opened for alternatives.

This approach is to have backward compatibility and ability to use `hydra:template` on both `hydra:IriTemplate` and newly introduced `hydra:HeaderTemplate`.

Another approach would be to introduce completely new `hydra:template` alternative for `hydra:HeaderTemplate`. I think it would be prone to errors.

Yet another approach would be to completely remove header templates at all. I believe templates could cover more complex header value scenarios, i.e. those for `PREFER`, where you provide some complex expressions that could be bound to variables.